### ISSUE TITLE
PostGIS Not Working

### ISSUE THREAD

**Issue by sstubbs** [yugabyte/yugabyte-db#8900]
I've installed postgis in the docker image  with the following docker file:
```
FROM yugabytedb/yugabyte:2.7.1.1-b1

RUN rpm -Uvh https://yum.postgresql.org/11/redhat/rhel-7-x86_64/pgdg-redhat-repo-latest.noarch.rpm
RUN yum update -y
RUN yum install -y postgresql11-server postgis31_11

RUN cp -v "$(/usr/pgsql-11/bin/pg_config --pkglibdir)"/*postgis*.so "$(/home/yugabyte/postgres/bin/pg_config --pkglibdir)"
RUN cp -v "$(/usr/pgsql-11/bin/pg_config --sharedir)"/extension/*postgis*.sql "$(/home/yugabyte/postgres/bin/pg_config --sharedir)"/extension
RUN cp -v "$(/usr/pgsql-11/bin/pg_config --sharedir)"/extension/*postgis*.control "$(/home/yugabyte/postgres/bin/pg_config --sharedir)"/extension

RUN /home/yugabyte/bin/post_install.sh -e
```
I have tied versions 2.7.1.1-b1 and 2.4.4.0-b7 and here is the sql for the table:
```
CREATE EXTENSION IF NOT EXISTS postgis;
CREATE EXTENSION IF NOT EXISTS pgcrypto;

CREATE TABLE IF NOT EXISTS location
(
    id uuid DEFAULT gen_random_uuid(),
    country_code VARCHAR(2),
    postal_code VARCHAR(20),
    place_name VARCHAR(180),
    admin_name1 VARCHAR(100),
    admin_code1 VARCHAR(20),
    admin_name2 VARCHAR(100),
    admin_code2 VARCHAR(20),
    admin_name3 VARCHAR(100),
    admin_code3 VARCHAR(20),
    geog geography(POINT, 4326),
    accuracy SMALLINT,
    PRIMARY KEY (id)
);
```
and here is a test record. I have checked a lot of different records and always get the same issue.
```
INSERT INTO location (
country_code, postal_code, place_name, admin_name1, 
admin_code1, admin_name2, admin_code2, admin_name3, 
admin_code3, geog, accuracy) VALUES 
('GB', 'AL3 8QE', 'Slip End', 'England', 'ENG', 'Bedfordshire', '', 'Central Bedfordshire', 'E06000056', ST_MakePoint(-0.4474, 51.8479), 6);
```
Checking this in the official PostGIS docker container with
```
SELECT st_x(geog::geometry), st_y(geog::geometry) FROM location;
```
gives me
```
stx -0.4474
sty 51.8479
```
which is correct but Yugabyte gives me
```
stx -0.4474
sty 0
```
which is not

**Comment by andrei-mart** [comment]
It looks like a problem with the ST_MakePoint function.
```
yugabyte=# select ST_MakePoint(-0.4474, 51.8479);
                st_makepoint                
--------------------------------------------
 0101000000E0BE0E9C33A2DCBF0000000000000000
(1 row)
```
Note these trailing zeros.
Correct value obtained from vanilla Postgres/PostGIS 
```
postgres=# select ST_MakePoint(-0.4474, 51.8479);
                st_makepoint                
--------------------------------------------
 0101000000E0BE0E9C33A2DCBFA323B9FC87EC4940
(1 row)
```
Yugabyte stores geography correctly.
```
yugabyte=# update location set geog = ST_GeogFromWKB(decode('0101000020E6100000E0BE0E9C33A2DCBFA323B9FC87EC4940', 'hex'));
UPDATE 1
yugabyte=# SELECT st_x(geog::geometry), st_y(geog::geometry) FROM location;
  st_x   |  st_y   
---------+---------
 -0.4474 | 51.8479
(1 row)
``` 
Taking look into ST_MakePoint.

**Comment by andrei-mart** [comment]
The problem was introduced by #5408.
It was basically a backport from Postgres 12, it changes the way how external functions are called and breaks plugins compiled against Postgres 11.

**Comment by andrei-mart** [comment]
Fixed by #9208
